EIPS/call_gas_split.md

```diff
@@ -0,0 +1,64 @@
+---
+title: CALL_GAS_SPLIT - Setting gas split portion for EOF calls
+description: Opcode that allows splitting gas sent to calls
+author: Dragan Rakita (@rakita)
+discussions-to: <URL>
+status: Draft
+type: Standard Track 
+created: 2025-03-25
+---
+
+## Abstract
+
+This EIP introduces a new instruction that allows EOF contracts to control the portion of gas forwarded to callees without exposing exact gas values.
+
+## Motivation
+
+With the introduction of revamped CALL instructions in [EIP-7069](./eip-7069.md), direct gas limiting for calls is no longer possible. However, there are legitimate use cases where limiting the gas forwarded to callees is necessary.
+
+This EIP addresses this need by allowing contracts to specify a percentage of available gas to be forwarded, without exposing exact gas amounts. This maintains the benefits of gas non-observability while providing necessary control.
+
+A key problem with always forwarding all available gas is that a malicious callee could consume all gas during execution (even in case of a revert), potentially causing denial of service. By allowing the caller to restrict the forwarded gas to a specific portion, this risk can be mitigated.
+
+By default, the gas split is set to `63/64` to match the existing behavior defined in [EIP-150](./eip-150.md).
+
+With future multidimensional gas models, this instruction could potentially restrict all gas types.
+
+## Specification
+
+A new instruction `CALL_GAS_SPLIT` is introduced. This instruction:
+
+1. Pops one item from the stack representing a gas split ratio.
+2. Floors this value to `65535` (u16::MAX).
+3. Calculates the gas limit by using the **current** gas by spliting it by floored value as `split_gas=(gas*value)/65535`.And preserves this value until the end of the call. 
+4. The preserved value should be used in all `EXT*CALL` operations to set gas limit as `max(split_gas, gas_left)`.
+
+If this value is not set, the EVM interpreter MUST maintain the `63/64` rule that was previously used as default.
```

Comment:
I'm assuming that the gas forwarded is still subject to the 63/64 rule. If this would not be the case, then the call stack depth limit of 1024 can be reached (this is also a DoS vector being able to cheaply create new call frames)

Reply:
It was briefly mentioned that 63/64 rule stayed in EOF calls to not disrupt the current pattern and have yet another test vector that we need to test. Having discussion on lifting this would be beneficial, I don't have strong opinion on it rn.

Reply:
Have added a line that this is up for discussion, it really depends on implementations if they can support big call stack, if it becomes troublesome, we can remove it.